File: Models/DTOs/CompanyDTO.cs

using System.ComponentModel.DataAnnotations;


namespace GoWheels_WebAPI.Models.DTOs
{
    public class CompanyDTO
    {
        public int Id { get; set; }
        [Required]
        public string? Name { get; set; }
        [Required]
        public string? IconImage { get; set; }
        public List<int> CarTypeIds { get; set; } = new List<int>();
    }
}

File: FE/src/DTOs/CompanyDto.js
// src/dtos/CompanyDTO.js

export default class CompanyDTO {
  constructor(data = {}) {
    this.id = data.id || 0;
    this.name = data.name || "";
    this.iconImage = data.iconImage || null; // File hình ảnh
    this.carTypeIds = data.carTypeIds || [];
  }

  toFormData() {
    const formData = new FormData();
    if (this.id !== null) {
      // Kiểm tra xem `id` có khác null không
      formData.append("Id", this.id);
    }
    formData.append("Name", this.name);
    if (this.iconImage) {
      formData.append("IconImage", this.iconImage); // File hình ảnh
    }
    this.carTypeIds.forEach((carTypeId) =>
      formData.append("CarTypeIds", carTypeId)
    );
    return formData;
  }
}
